plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1183,18 +1191,46 @@ def get_transaction_receipt(
         "Error when attempting getting tx receipt: {}", logger_method="debug"
     )
     def _try_get_transaction_receipt(
-        self, tx_digest: str, **_kwargs: Any
+        self,
+        tx_digest: str,
+        timeout: Optional[int] = None,
+        poll_latency: Optional[float] = None,
+        **_kwargs: Any,
     ) -> Optional[JSONLike]:
         """
         Try get the transaction receipt.
 
         :param tx_digest: the digest associated to the transaction.
+        :param timeout: time to wait before assuming the receipt is not available, defaults to infinite.
+        :param poll_latency: backoff between retries, if None, a single try (call) will be made.
         :param _kwargs: the keyword arguments. Possible kwargs are:
             `raise_on_try`: bool flag specifying whether the method will raise or log on error (used by `try_decorator`)
         :return: the tx receipt, if present
         """
-        tx_receipt = self._api.eth.get_transaction_receipt(  # pylint: disable=no-member
-            cast(HexStr, tx_digest)
+        if timeout is None and poll_latency is None:
+            # if timeout and poll_latency have not been provided
+            # we make a single request
+            tx_receipt = (
+                self._api.eth.get_transaction_receipt(  # pylint: disable=no-member
+                    cast(HexStr, tx_digest),
+                )
+            )
+            return AttributeDictTranslator.to_dict(tx_receipt)
+
+        # either a polling interval or a timeout was provided
+        if timeout is None:
+            # if not provided, we wait for ~31 years.
+            timeout = 10**9
+        if poll_latency is None:
+            # if not provided, we don't make multiple retires.
+            poll_latency = float(timeout)
+
+        tx_receipt = (
+            self._api.eth.wait_for_transaction_receipt(  # pylint: disable=no-member
+                cast(HexStr, tx_digest),
+                timeout=timeout,
+                poll_latency=poll_latency,
+            )
```

Comment:
@DavidMinarsch , @Adamantios discovered that this doesn't solve our problem, i.e. blocking requests.
The timeout works only if the requests don't block. Which is not what we expected. 

I suggest that we move this problem back to the ledger connection. We can have a generic `with_timeout()` method that can wrap any of these api methods into a coroutine, and then we have a real timeout mechanism.
Furthermore, that will allow us to make sure that any of the other requests doesnt block as well.

Reply:
Yes agree- let's solve it on that level